tensorizer/serialization.py

```diff
@@ -432,11 +432,23 @@ def _read_shapes(obj, num_elems) -> List[int]:
 
     @property
     def total_bytes_read(self) -> int:
+        if hasattr(self._file, "bytes_read"):
+            return self._file.bytes_read
         if self._file.closed:
             return self.total_tensor_bytes
         else:
             return self._file.tell()
 
+    # If our _file object has 'response_headers' attribute, we can use it to
+    # to get the total size of the file, in addition to whether the object
+    # was cached.
+    @property
+    def is_cached(self) -> bool:
+        if hasattr(self._file, "response_headers"):
+            return self._file.response_headers.get("X-Cache-Status", "MISS") == "HIT"
```

Comment:
there are OTHER values beyond HIT/MISS, can you please capture & expose them?

Reply:
Hm. They're indirectly available with `response_headers`. Do we want to return a string instead of a `boolean`?

Reply:
yes please. I suspect it will be more important on the problematic requests than the requests that are working well. (e.g. why did it miss, was it revalidating etc)

Reply:
OK, changing to `cache_status` that returns either `False` or the string value. That good?